src/Controller/V2/Quarterback/QuarterbackController.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * © 2023 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Manuscript } from '@manuscripts/json-schema'
+import { Request } from 'express'
+
+import { DIContainer } from '../../../DIContainer/DIContainer'
+import { ProjectPermission } from '../../../DomainServices/ProjectService'
+import { RoleDoesNotPermitOperationError, ValidationError } from '../../../Errors'
+import { Snapshot } from '../../../Models/SnapshotModel'
+import { ContainedBaseController } from '../../ContainedBaseController'
+
+export class QuarterbackController extends ContainedBaseController {
+  async getPermissions(projectID: string, userID: string): Promise<ReadonlySet<ProjectPermission>> {
+    return DIContainer.sharedContainer.projectService.getPermissions(projectID, userID)
```

Comment:
Thinking about this more, using the project permissions might not work. For example, an annotator will not be allowed to create a manuscript (or even update an existing one), but they should have access to update a doc or create a snapshot.

Reply:
I'm thinking of creating a new set of permissions that should cover all action on quarterback, Is there a place where these permissions are well defined?

maybe for now and until we have the delta changes @mbartenev-atypon is working on, we can assign READ & WRITE permission for all roles (except readers which will have read only writes), then we can refine it better.

Reply:
Yes, let's do that (write only disabled for viewers).